coderd/httpmw/logger.go

```diff
@@ -74,3 +85,37 @@ func Logger(log slog.Logger) func(next http.Handler) http.Handler {
 		})
 	}
 }
+
+type RequestLoggerContext struct {
+	log     *slog.Logger
+	written bool
+	message string
+}
+
+func (c *RequestLoggerContext) WithFields(fields ...slog.Field) {
+	newLogger := c.log.With(fields...)
+	c.log = &newLogger
+}
```

Comment:
I don't think the pointer for `*slog.Logger` is required, should be fine to just do `c.log = c.log.With(fields...)`